src/app/calculation.cpp

```diff
@@ -85,268 +97,356 @@ QString Calculation::getError()
+
+            // old: 0.89F * (1-vgs) * fsant * R4V;
+            infiltrationFromPerviousRoads =
+                (1 - record.imperviousnessRoad) *
+                areaShareRoads * bagrovSurfaceClass4;
 
             // runoff for unsealed surfaces rowuv = 0
-            riuv = (100.0F - (float) ptrDA.VER) / 100.0F * RUV;
+            infiltrationFromPerviousSurfaces = (
+                100.0F - (float) resultRecord.imperviousness
+            ) / 100.0F * unsealedSurfaceRunoff;
+
+            // calculate runoff 'row' for entire block patial area (FLGES +
+            // STR_FLGES) (mm/a)
+            runoff = (
+                runoffSealedSurface1 +
+                runoffSealedSurface2 +
+                runoffSealedSurface3 +
+                runoffSealedSurface4 +
+                runoffRoofs +
+                runoffPerviousRoads
+            );
 
-            // calculate runoff 'row' for entire block patial area (FLGES+STR_FLGES)
-            row = (row1 + row2 + row3 + row4 + rowd + rowuvs); // mm/a
-            ptrDA.ROW = INT_ROUND(row);
+            resultRecord.runoff = INT_ROUND(runoff);
             
-            // calculate volume 'rowvol' from runoff
-            ROWVOL = row * 3.171F * (fb + fs) / 100000.0F;     // qcm/s
+            // calculate volume 'rowvol' from runoff (qcm/s)
+            rainwaterRunoff = runoff * 3.171F * (
+                record.totalAreaBuildings +
+                record.totalAreaRoads
+            ) / 100000.0F;
             
             // calculate infiltration rate 'ri' for entire block partial area
-            ri = (ri1 + ri2 + ri3 + ri4 + rid + riuvs + riuv); // mm/a
-            ptrDA.RI = INT_ROUND(ri);
-            
-            // calculate volume 'rivol' from infiltration rate
-            RIVOL = ri * 3.171F * (fb + fs) / 100000.0F;       // qcm/s
+            // (mm/a)
+            infiltrationRate = (
+                infiltrationFromSealedSurface1 +
+                infiltrationFromSealedSurface2 +
+                infiltrationFromSealedSurface3 +
+                infiltrationFromSealedSurface4 +
+                infiltrationFromRoofs +
+                infiltrationFromPerviousRoads +
+                infiltrationFromPerviousSurfaces
+            );
+
+            resultRecord.infiltrationRate = INT_ROUND(infiltrationRate);
             
-            // calculate total system losses 'r' due to runoff and infiltration for entire block partial area
-            r = row + ri;
-            ptrDA.R = INT_ROUND(r);
+            // calculate volume 'rivol' from infiltration rate (qcm/s)
+            totalSubsurfaceFlow = infiltrationRate * 3.171F * (
+                record.totalAreaBuildings +
+                record.totalAreaRoads
+            ) / 100000.0F;
+
+            // calculate total system losses 'r' due to runoff and infiltration
+            // for entire block partial area
+            totalSystemLosses = runoff + infiltrationRate;
```

Comment:
I am not quite sure, why 'r' is called 'total system losses' in comments? I would rather call it total runoff (as named in abimo report) and 'row' would be surfaceRunoff

Reply:
Also, I have doubts about the word 'Rate' in infiltrationRate. Since both variables 'runoff' and 'infiltrationRate' have the same unit (mm), it might be a bit confusing to have 'rate' only for one.

Reply:
totalRunoff = surfaceRunoff + infiltration

Reply:
Great proposition. Done via 6ec5637926f283d818d45c9b611f10c304b4cd7b.